docs/02-guides/server-side-events/create-an-event.mdx

```diff
@@ -7,9 +7,119 @@ sidebar_position: 1
 
 <p>{frontMatter.description}</p>
 
-:::info WIP
+## What is an event emitter
 
-- Create an event emitter
-- Emit an event from your code
+The events are dispatched with what we call Event Emitter. The role of the event emitter is to push events to the Server event pipeline to be processed by the server.
 
-:::
+In this guide, we will cover the creation of an emitter and how to implement them in your pages, we will create an Emitter that Emit an event when a customer reach a page called FAQ.
+
+## Create a new route
+
+Create the following route in your project:
+
+```tsx title="app/routes/_main.faq.tsx"
+export default function FAQ() {
+  return (
+    <div>
+      <h1>FAQ</h1>
+    </div>
+  );
+}
+```
+
+And now test your route by starting your Front-Commerce server and navigate to [http://localhost:4000/faq](http://localhost:4000/faq)
+
+You should now see your brand new page
+
+## Create an Emitter class
+
+Let's create a file in our project, called 'FAQPageReached':
+
+```tsx title="app/server-events/events/FAQPageReached.ts"
+import { ServerEvent } from "@front-commerce/core/events";
+
+// Name of the event that will be dispatched to the external services
+const FAQPageReachedName = "FAQPageReachedEvent";
+
+// Object that will be sent to external services
+type FAQPageReachedEventType = {
+  question: string;
+};
+
+export default class FAQPageReachedEvent extends ServerEvent<FAQPageReachedEventType> {
+  static event_type = FAQPageReachedName;
+
+  // In our guide we will check if customer pass
+  // And we will submit a metadata object containing metadatas about the event
+  // Here we will retrieve the locale
+  constructor(question: string, locale: string) {
+    // Call the constructor of the base class, the new Date() match the event creation date
+    super(FAQPageReachedName, new Date(), { question }, { locale });
+  }
+}
+```
+
+That's it!
+
+## Use the Emitter
+
+Now that we've created an Emitter, let's use it in our page:
+
+```tsx title="app/routes/_main.faq.tsx"
+import { FrontCommerceApp } from "@front-commerce/remix";
+import FAQPageReachedEvent from "../server-events/events/FAQPageReachedEvent";
+import type { LoaderFunctionArgs } from "@remix-run/node";
+
+//highlight-start
+export async function loader({ context, request }: LoaderFunctionArgs) {
+  // We create a Front-Commerce App to access the ServerEventBus Service
+  const app = new FrontCommerceApp(context.frontCommerce);
+  // We parse the question from the URL
+  const question = new URL(request.url).searchParams.get("question");
```

Comment:
**question:** As of right now, there is nothing in the code that will set any `question` params. Should this example re-use the existing `skeleton/example-extensions/faq-demo` instead, so that it works out of the box?

**note:** It would also require updating this guide a bit.

Reply:
This example is using simple GET searchParam and inject it in event, the mail goal is to tell the user "see, you can inject custom stuff in it".
I don't think it's necessary to go further

Reply:
I think it's important to have working examples, as @Kagaminara mentions there is no working example here, it's a half implemented feature.

As an integrator I would expect to be able to copy paste this in my application, then it should work.

Reply:
I would even argue that it should be added to the faq-demo: https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/b6381e9b4daf14a321aec1f3e45de064e7504160/skeleton/example-extensions/faq-demo/routes

Reply:
Added to FAQ demo, please see relevant MR in main repo

Reply:
Resolving as the FAQ demo is now used / mentioned in this guide.